addons/mail/wizard/mail_compose_message.py

```diff
@@ -807,7 +809,8 @@ def _prepare_mail_values(self, res_ids):
 
         if email_mode:
             mail_values_all = self._process_mail_values_state(mail_values_all)
-        return mail_values_all
+        return (mail_values_all if not email_mode or mass_include_canceled
+                else {res_id: values for res_id, values in mail_values_all.items() if values.get('state') != 'cancel'})
```

Comment:
Not creating canceled emails doesn't sound like a bad idea, but I think overall for this task we might be better served by adding a filter to `message_fetch` to hide canceled messages that don't have notifications. It seems like that would be doable with a simple domain, which shouldn't make `message_fetch` much costlier.

I see you've investigated the same (or a similar) approach but _message_fetch already does a query anyway, so doesn't just updating that domain work?

I'm also mildly worried the current approach might mess with statistics but it's probably not a real issue.

Reply:
Ok, I see your point. Indeed it might be best not to create the canceled ones. It's not like we can correct/resend them anyway so the usability impact is minor to none.

Let's go with this then :)